packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -740,11 +789,27 @@ export class GarbageCollector implements IGarbageCollector {
         // Basically, it was written in the current GC version.
         if (result.wasSummaryTracked) {
             this.latestSummaryGCVersion = this.currentGCVersion;
+            this.latestSerializedGCBlobs = this.pendingSerializedGCBlobs;
             return;
         }
-        // If the summary was not tracked by this client, update latest GC version from the snapshot in the result as
-        // that is now the latest summary.
-        await this.updateSummaryGCVersionFromSnapshot(result.snapshot, readAndParseBlob);
+        // If the summary was not tracked by this client, update latest GC version and blob from the snapshot in the
+        // result as that is now the latest summary.
+        const snapshot = result.snapshot;
+        const metadataBlobId = snapshot.blobs[metadataBlobName];
+        if (metadataBlobId) {
+            const metadata = await readAndParseBlob<IContainerRuntimeMetadata>(metadataBlobId);
+            this.latestSummaryGCVersion = getGCVersion(metadata);
+        }
+
+        const gcSnapshotTree = snapshot.trees[gcTreeKey];
+        if (gcSnapshotTree !== undefined && this.trackGCBlobState) {
```

Comment:
If `gcSnapshotTree` is undefined, the latest GC blobs would now point to the previous latest summary's state and not the one for which we received an ack.

Reply:
What cases are `gcSnapshotTree === undefined` here?

Reply:
A summarizer that doesn't write GC data at root which is still not the default. But irrespective of that, since there is such a scenario possible in code (maybe because of a bug?), it has to be handled. For instance, this should either assert that `gcSnapshotTree` cannot be undefined or handle the case where it can be undefined.